src/v/cluster/cluster_utils.cc

```diff
@@ -20,35 +20,39 @@
 
```

Comment:
> Fixed calculating added/deleted brokers lists. Previously algorithm was
using all brokers field to compare brokers, currently we skip
comparing membership state when comparing brokers.

can you add: _why_ it was broken and now correct? future readers could mostly deduce what's in the commit message from looking at the code.